x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -47,18 +48,39 @@
         Setting.Property.NodeScope
     );
 
+    // Deliberately unregistered, we only use this in tests
+    public static final Setting<Boolean> DELETE_ON_EVICTION_SETTING = Setting.boolSetting(
```

Comment:
I've made a suggestion in Slack how we can possibly avoid introducing this setting. The idea is to simulate a crash by blocking write access to the disk during a restart, similar as is done in DiskDisruptionIT.